### ISSUE TITLE
netstandard2 support

### ISSUE THREAD

**Issue by arctouch-oberdanbitencourt** [urbanairship/urbanairship-xamarin#90]
Hi. I'm porting a large application I built that uses the UrbanAirship.Portable nuget package to netstandard2.

Currently my app is unable to resolve AirshipBindings.Portable and I suspect that it is because the package is built for PCL and does not support yet netstandard. I tried setting the `<AssetTargetFallback>portable-net45+wp8+wpa81+win8+MonoAndroid41+MonoTouch10+Xamarin.iOS10</AssetTargetFallback>` in my netstandard csproj file, but it didn't help.

Do you have plans to releasing a new version supporting netstandard?

**Comment by rlepinski** [comment]
So this is actually a new thing to us - https://docs.microsoft.com/en-us/xamarin/xamarin-forms/internals/net-standard

We will take a look

**Comment by arctouch-oberdanbitencourt** [comment]
Hi @rlepinski. I bring good news (I hope) to you.

I took the liberty of cloning this project and ported the UrbanAirship.Portable project to netstandard, fixed the references and then referenced it directly in my project. I was able to get push notifications working on my iOS app, however I couldn't test Messaging Center due to dependencies in my code I need to fix beforehand.

The good news is that the migration is pretty simple and I didn't stumble upon anything in the middle of the road. To understand better how to do it, I suggest reading this post: https://montemagno.com/how-to-convert-a-pcl-library-to-net-standard-and-keep-git-history/

Basically I replaced the whole content for `AirshipBindings.Portable.Abstractions.csproj` and `AirshipBindings.Portable.csproj` to
```xml
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFrameworks>netstandard2.0</TargetFrameworks>
</PropertyGroup>
```
and referenced `AirshipBindings.Portable.Abstractions` on the `AirshipBindings.Portable` project. I also deleted the `AssemblyInfo.cs` file as it is no longer used for netstandard projects (you can migrate the content there to the .csproj file).

You guys may want to discuss internally if you still want to support PCL profiles for future releases.

**Comment by rlepinski** [comment]
Thanks for the info. We are looking into adding it but it may take us a week or so due to other priorities.

**Comment by arctouch-oberdanbitencourt** [comment]
Thanks a lot @rlepinski. I'm looking forward to trying out your netstandard package.

**Comment by rlepinski** [comment]
@arctouch-oberdanbitencourt https://github.com/urbanairship/xamarin-component/pull/92
It looks ready to go, its just waiting on reviews from my team. Most likely it will be released by the end of the week.

**Comment by arctouch-oberdanbitencourt** [comment]
Ah, I should've looked at the open PRs. Outstanding! Thanks for the info.

**Comment by marc-scig** [comment]
Hey @arctouch-oberdanbitencourt ,

5.0.2 is now out, which includes a .NET standard 2.0 package on nuget.  Functionally it's pretty much identical to the PCL, so hopefully it should be pretty straightforward. Give it a try, and if you have any feedback don't hesitate to let us know.

**Comment by arctouch-fabriciomiranda** [comment]
Hello @marc-scig I've been working with @arctouch-oberdanbitencourt in this migration and we encountered a while trying to run an iOS application with the new .NET standard 2.0 package:

```
2018-06-18 17:05:46.588 xxxxxx.iOS.xxxx[25354:1319052] Could not find `AirshipBindings.NETStandard` referenced by assembly `xxxxxxxx, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null`.
```

With the same references (urbanairship.netstandard nuget and urbanairship.android.core) in Android everything seems to work fine, but not in iOS, the application is unable to find the Bindings for some unknown reason, I've tried to add the respective NuGet packages (urbanairship.netstandard and rbanairship.ios) to the iOS project, but it still fails at runtime, any thoughts on this? (all version 9.0.0)

**Comment by BrianBatchelder** [comment]
I am unable to reproduce this at my desk. It could be you have some cruft in your project file left over from building with our SDK pre 9.0. Here are some ideas:
- Do a _really_ clean build, i.e. manually remove all `bin` and `obj` directories.
- Check your .csproj files:
   - Remove any references to `AirshipBindings.NETStandard` and `AirshipBindings.NETStandard.Abstractions`.
   - For the root and iOS .csproj, make sure your only PackageReferences are to `urbanairship.netstandard`. For Android, you will also need one or more push providers, e.g. `urbanairship.android.fcm`.
- Remove and re-add the urbanairship nuget packages.

Please let us know if this solves your issue.

**Comment by BrianBatchelder** [comment]
@arctouch-fabriciomiranda @arctouch-oberdanbitencourt 
Closing this issue. Please re-open if you have further issues.
